DecipherNow/helm-charts
# Secrets refactor

### Updates
- Removes all secret files from Grey Matter charts and moves them to a new "helper" chart called `secrets`. The `values.yaml` file for this chart is essentially `greymatter-secrets.yaml`, and will template out all necessary secrets referenced in GM charts. This is an *optional* chart that will help with first time setup of the mesh.
- Update all charts to use a new secret format. Now each deployment has a `secret` block (see below.) The `secretKeys` reference the keys used in the secret, and are also used as the filenames.
```yaml
  secret:
    secret_name: sidecar-certs
    mountPoint: /etc/proxy/tls/sidecar
    secretKeys:
      ca: ca.crt
      cert: service.crt
      key: service.key
```
### Testing
These installs assume helm@3
1. Install secrets:
```
helm install secrets secrets -f credentials.yaml
```
2. Install fabric
```
rm -rf fabric/charts && helm dep up fabric
helm install fabric fabric --set=global.environment=kubernetes
```
3. Install data
```
rm -rf data/charts && helm dep up data
helm install data data --set=global.environment=kubernetes
```
4. Install sense
```
rm -rf sense/charts && helm dep up sense
helm install sense sense --set=global.environment=kubernetes
```
5. Install voyager
```
helm repo add appscode https://charts.appscode.com/stable/
helm repo update
helm install voyager-operator appscode/voyager --version v12.0.0-rc.1 \
  --namespace kube-system \
  --set cloudProvider=minikube
```
6. Install edge
```
helm install edge edge --set=global.environment=kubernetes
```

You should see the following. Note that catalog will not come up because it relies on GM routing, which we haven't set up yet.
```
NAME                                     READY   STATUS             RESTARTS   AGE
catalog-d446d67-gcctl                    1/2     CrashLoopBackOff   15         55m
catalog-init-w7q8l                       0/1     Init:0/1           0          55m
control-687dd8cd6-88rtt                  1/1     Running            0          56m
dashboard-5975d8d95d-mg74c               2/2     Running            0          55m
data-0                                   2/2     Running            0          55m
data-internal-0                          2/2     Running            0          55m
data-mongo-0                             1/1     Running            0          56m
gm-control-api-0                         2/2     Running            0          55m
internal-data-mongo-0                    1/1     Running            0          56m
internal-jwt-security-5c86c69b5d-sm98r   2/2     Running            1          56m
internal-redis-6685956c57-nb77m          1/1     Running            0          56m
jwt-security-75c459958b-hdm5m            2/2     Running            2          56m
postgres-slo-0                           1/1     Running            0          55m
prometheus-0                             2/2     Running            0          55m
redis-57c677dd7f-9rvqr                   1/1     Running            0          56m
slo-754c6d5667-2wn8x                     2/2     Running            0          55m
edge-6c8664c66b-pjgvv                    1/1     Running            0          2s
```
